### PR TITLE
chore: publish gce images with releases

### PR DESCRIPTION
**Pull request by rsmitty**
This is somewhat tough to test, since you can't use the github release for anything but tags (it throws an error), but this PR will build the GCE image and output it to the build directory on tag events. It should, then, get added to the release assets. Once merged, I'll push a test tag to make sure the draft looks correct.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -172,8 +172,8 @@ proto: buildkitd
 .PHONY: talos-gce
 talos-gce:
 	@docker run --rm -v /dev:/dev -v $(PWD)/build:/out --privileged $(DOCKER_ARGS) autonomy/installer:$(TAG) install -n disk -r -p googlecloud -u none
-	@tar -C $(PWD)/build -Sczf $(PWD)/build/$@.tar.gz disk.raw
-	@rm $(PWD)/build/disk.raw
+	@tar -C $(PWD)/build -czf $(PWD)/build/$@.tar.gz disk.raw
```
Comment (rsmitty) [inline Makefile]: For some reason, the tar that's in the drone container doesn't have the sparse (-S) option. It's not a massive difference anyways, looked like 10MB in file size locally.

File: Makefile [hunk 2]
```diff
@@ -172,8 +172,8 @@ proto: buildkitd
 .PHONY: talos-gce
 talos-gce:
 	@docker run --rm -v /dev:/dev -v $(PWD)/build:/out --privileged $(DOCKER_ARGS) autonomy/installer:$(TAG) install -n disk -r -p googlecloud -u none
-	@tar -C $(PWD)/build -Sczf $(PWD)/build/$@.tar.gz disk.raw
```
Comment (andrewrynhard) [inline Makefile]: I think we want to keep the `-S` so we aren't tarring up a bunch of zeros. We will need to update the build-container with an `apk add tar`, the busy box implementation doesn't support `-S` it seems.

File: .drone.yml [hunk 3]
```diff
@@ -179,7 +196,8 @@ steps:
     depends_on:
       ## Should change to e2e once we get things more stable
       - basic-integration
-      - iso
+     - iso
+      - gce
```
Comment (andrewrynhard) [inline .drone.yml]: nit: Align these.

File: Makefile (hunk 2 above)
Comment (rsmitty) [reply Makefile]: See comment above. For some reason we don't have the -S or --sparse flag in the tar binary that the drone runner has installed.

Comment (andrewrynhard) [reply Makefile]: its busy box implementation :/

**Pull request merged**